core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -2285,6 +2285,7 @@ private SqlNode registerFrom(
     case VALUES:
     case WITH:
     case OTHER_FUNCTION:
+    case TUMBLE:
```

Comment:
This is not necessary, handle the function like lateral table [1]

[1] https://github.com/apache/calcite/blob/b93ec5a9edb7459696385e6adad67b92b6d406d7/core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java#L2256

Reply:
I believe in Calcite `COLLECTION_TABLE` is the `TABLE(...)` part in `SELECT * FROM TABLE(TUMBLE(...))`. 

I will give it try to change TUMBLE's sql kind to `COLLECTION_TABLE` tomorrow and see how will it work.

Reply:
Ok. So I think the suggestion to use `COLLECITON_TABLE` is to address the special handling in validator. I gave it a try to make `TUMBLE`'s SqlKind as `COLLECITON_TABLE` and found

1. most of the special handling for TUMBLE is not needed
2. except for 
```
    if (rexCall.getKind() == SqlKind.TUMBLE) {
      inputs.add(convertSelect((SqlSelect) call.getOperandList().get(0), false));
    } else {
      inputs = bb.retrieveCursors();
    }
```
as inputs is still empty by using `COLLECITON_TABLE`.

3. In enumerable implementation, string match is used to check if it is tumbling case as we no longer can use SqlKind to check it, which makes code a little longer.


Because the inputs is still empty problem, I incline to still use SqlKind.TUMBLE.